src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpPreferAsSpanOverSubstring.Fixer.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.NetCore.Analyzers.Runtime;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Runtime
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public sealed class CSharpPreferAsSpanOverSubstringFixer : PreferAsSpanOverSubstringFixer
+    {
+        private protected override void ReplaceInvocationMethodName(SyntaxEditor editor, SyntaxNode memberInvocation, string newName)
+        {
+            var cast = (InvocationExpressionSyntax)memberInvocation;
+            var memberAccessSyntax = (MemberAccessExpressionSyntax)cast.Expression;
```

Comment:
Are we sure we can't be on something else than `MemberAccessExpresionSyntax` here? Maybe a member binding or something like this.

Reply:
The only way this could be a `MemberBindingExpressionSyntax` is if the substring invocation is conditional, `foo?.Substring(1)`. 
This can't possibly be true because the analyzer detects `IArgumentOperation`s, and after walking down implicit conversions, the syntax of `foo?.Substring(1)` would be `ConditionalAccessExpressionSyntax`, and not `InvocationExpressionSyntax`.
I'll add some assertions and comments to enforce this invariant.